src/gdm/temporal_models.py

```diff
@@ -0,0 +1,39 @@
+from datetime import date
+
+from gdm import DistributionComponentBase
+from pydantic import field_validator
+from infrasys import Component
+from uuid import UUID
+
+
+class DeletionModel(Component):
```

Comment:
Why are these subtypes of Component? Also, a `Component` already has the fields `name` and `uuid`, so this could be confusing.